.github/workflows/CI.yml

```
@@ -274,3 +274,30 @@ jobs:
         run: |
           SET CHERE_INVOKING=1
           C:\cygwin64\bin\bash -l -c "bash dev/ci.sh"
+
+  slack-notification-on-failure:
+    name: Send slack notification on CI failure
+    needs:
+      - test-unix
+      - test-cygwin
+    if: ${{ always() && github.event_name == 'push' && github.repository == 'ssiccha/gap' }}
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Get branch name
+        id: get-branch
+        run: echo ::set-output name=branch::${GITHUB_REF#refs/*/}
+      - name: Determine whether CI status changed
+        uses: Jimdo/should-i-notify-action@0f48901c482c651cba9dc19a03efc068b402789b
+        id: should_notify
+        with:
+          branch: ${{ steps.vars.outputs.branch }}
+          needs_context: ${{ toJson(needs) }}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Send slack notification
+        uses: act10ns/slack@e4e71685b9b239384b0f676a63c32367f59c2522
+        if: steps.should_notify.outputs.should_send_message == 'yes'
```

This doesn't seem to work (two failed jobs in a row at https://github.com/ssiccha/gap/actions both sent the message) - perhaps it needs this syntax?
```suggestion
        if: ${{ steps.should_notify.outputs.should_send_message == 'yes' }}
```

---

Two succeeded jobs in a row didn't sent a message though. So in principle it works I think. OTOH the action https://github.com/Jimdo/should-i-notify-action states in its README that it always returns `yes`, if the action failed. So I guess we have to extend or change that. I've checked it out and the action itself is not too complicated. I have an appointment now though, so I'll have to get back to this another time.

---

Ah I didn't realise. Thanks for the clarification.

---

I've now made a fork of the should-i-notify-action. See my new post below.